docs/polkadot/networking/3-attestation.md

```diff
@@ -1,54 +1,268 @@
+
+We rely on an underlying gossip network that allows us to broadcast various metadata to everyone, namely:
+
+1. receipts of specific pieces ("I have X piece")
+2. receipts of "enough" pieces ("I have >1/3 of pieces, I don't need more"). This could also double as an attestation to everyone else, that there is availability from the attester's position on the network.
+3. [TBX S1 #1]
+
+This metadata should be gossiped every few seconds. The data of the actual pieces are distributed via a separate topology as described below:
+
+Recall that we have a disjoint partition of N validators into C sets of parachain validators, each set having size N/C. For our purposes for this subprotocol, we will randomly assign a co-ordinate (c, i) to every validator, with c in [0, C) and i in [0, N/C). Fixing c and varying i defines a particular parachain validator set; varying c and fixing i defines what we'll call a particular validator "ring". This name is only meant to be very slightly suggestive, the precise structure and its justification will be described below.
+
+A validator ring is mostly-connected as permitted by the physical topology. Nodes within this ring talk to each other periodically via short-term and low-cost QUIC connections.
+
+For a given parachain, the primary checkers are also mostly-connected, as permitted by the physical topology. Likewise the secondary checkers are also mostly-connected.
+
+These connections represent the vast majority of traffic flow in our A&V networking protocol; to improve reliability and availability there are also other lines of communication as described below.
+
+The protocol runs in several phases and stages. Every node acts both in the distributor and distributee role, but not every role is active in every stage. A summary follows:
+
+Phase | Distributors | Distributees
+----- | ------------ | ------------
+P1SA  | Y            | N
+P1SB  | Y            | Y
+P2SA  | Y            | N
+P2SB  | N            | Y
+
+Note that there is also background activity, as described below.
+
+## Protocol phase 1: initial distribution
+
+Every validator is both a distributor of C pieces and a distributee (recipient) of C pieces [TODO: correction, explanation]. Every piece has one source parachain and one main target storer, and so we can index pieces with a tuple (c_s, (c_t, i_t)).
+
+In phase 1, pieces are distributed by the source parachain validators (c_s, \*) to the main target storers. This happens in two stages. Stage A is where most of the material is distributed, and stage B acts as a backup mechanism for anything that was missed during stage A.
+
+**Stage A**
+
+As a distributor, each validator (c, i) attempts to send the relevant pieces for their parachain's block (c, (c', i)), to everyone else on their ring (c', i) for all c' != c. Conversely as a distributee, each validator expects to receive their relevant pieces (c', (c, i)) for other parachains' blocks, from everyone else on their ring (c', i) for all c' != c.
```

Comment:
> to everyone else on their ring (c', i) for all c' != c.

If I understood the concept of a ring correctly, there is ever only one validator per parachain on such ring. Thus `c' != c` is not needed here, right?

Reply:
It's implied, but I thought it was clearer to be explicit about it.

Reply:
:+1: good to know that I am not missing something. Let's leave it as it is.